packages/blockchain-gateway-v1/src/FabricConnection.ts

```diff
@@ -168,38 +173,29 @@ export abstract class FabricConnection {
             eventHandlerOptions: { commitTimeout: timeout }
         };
 
-        await this.gateway.connect(this.connectionProfilePath, options);
-    }
+        await this.gateway.connect(connectionProfile, options);
 
-    protected async getChannel(channelName: string): Promise<Client.Channel> {
-        const client: Client = this.gateway.getClient();
-        let channel: Client.Channel = client.getChannel(channelName, false);
-        if (channel) {
-            return channel;
-        }
-        channel = client.newChannel(channelName);
-        const peers: Client.Peer[] = this.getAllPeers();
-        let lastError: Error = new Error(`Could not discover information for channel ${channelName} from known peers`);
-        for (const target of peers) {
-            try {
-                await channel.initialize({ asLocalhost: this.discoveryAsLocalhost, discover: this.discoveryEnabled, target });
-                return channel;
-            } catch (error) {
-                lastError = error;
-            }
+        // This bit is needed to add all the peers to the list of peers the lifecycle knows about
```

Comment:
Do we need to do this before connecting?
Do we call this code because 'lifecycle' doesn't maintain a list of peers? Is this what they do in the sdk?

Reply:
no we do it after connecting so that the gateway has set everything up. We need it because we need to tell lifecycle about all the peers so we can use it to later get all the committed smart contracts and all the channels.